packages/core/src/lib/code-snippet.svelte

```diff
@@ -0,0 +1,260 @@
+<!--
+@component
+  
+For rendering readonly blocks of code. Uses `prism` for syntax highlighting:
+
+https://prismjs.com
+
+```svelte
+<Breadcrumbs crumbs={['Dessert', 'Cookie']} />
+```
+-->
+<svelte:options immutable />
+
+<script
+  context="module"
+  lang="ts"
+>
+export type CodeSnippetTheme = 'vs' | 'vsc-dark-plus';
+
+// See: https://github.com/PrismJS/prism/releases
+const PRISM_VERSION = '1.29.0' as const;
+
+// See: https://github.com/PrismJS/prism-themes/releases
+const PRISM_THEMES_VERSION = '1.9.0' as const;
+
+const loadedLanguages: Record<string, boolean> = {};
+</script>
+
+<script lang="ts">
+import { createEventDispatcher, onMount } from 'svelte';
+import cx from 'classnames';
+
+import type { IconName } from '$lib';
+import IconButton from './button/icon-button.svelte';
+
+/**
+ * The language to use for syntax highlighting. Must be a language supported by
+ * prism: https://prismjs.com/#supported-languages
+ */
+export let language: string;
+
+/**
+ * The code to render in the code-snippet.
+ */
+export let code: string;
+
+/**
+ * The theme to use for syntax highlighting. Must be a theme provided by
+ * prism-themes: https://github.com/PrismJS/prism-themes#available-themes
+ *
+ * Currently only the `vs` and `vsc-dark-plus` themes are supported.
+ */
+export let theme: CodeSnippetTheme = 'vs';
+
+/**
+ * Whether or not to show a copy-to-clipboard button.
+ */
+export let showCopyButton = true;
+
+/**
+ * Some prism languages have dependencies. For example, C++ requires C. If the
+ * passed `language` has dependencies, they must be included here to be loaded.
+ */
+export let dependencies: string[] = [];
+
+/** Additional CSS classes to pass to the aside. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+let element: HTMLElement;
+
+$: copyButtonLabel = 'Copy';
+$: copyButtonIcon = 'content-copy' as IconName;
+
+const dispatch = createEventDispatcher<{
+  copy: { succeeded: boolean; message: string };
+}>();
+
+const getCdnUrl = (src: string) =>
+  `https://cdnjs.cloudflare.com/ajax/libs/prism/${PRISM_VERSION}/${src}`;
```

Comment:
Why aren't we using npm and importing the file(s) from `node_modules`? Relying on a CDN feels hacky

Reply:
This is how it was previously handled, I believe to allow loading the library asynchronously. @micheal-parks and @mrloureed  may know better.

Reply:
That was me back in the good ol' days. If I recall correctly I did this only to lazy load this library. Doing a dynamic import from node_modules would be much better and worth doing during this port.

Reply:
Cool, should clean things up then. What about the CDN for the CSS?

Reply:
Beware though, I might have done it this way because of prism not playing well with vite or something. I'd usually go for a node_modules / dynamic import solution.

Reply:
You can dynamically import css as well, you can probably get the same files from node modules: https://colinchjs.github.io/2023-10-13/09-59-05-131860-can-you-dynamically-import-css-files-using-javascripts-dynamic-imports/

Reply:
@mcous @micheal-parks I threw together an implementation using dynamic imports and leveraging the `autoloader` plugin for `prism`.

Reply:
This looks nice!